dev/tasks/r/github.nightly.yml

```diff
@@ -0,0 +1,381 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+{% import 'macros.jinja' as macros with context %}
+{% set r_root = "nightly/r" %}
+{% set webdav_domain = "https://nightly.wujciak.de:8080" %}
+{% set repo_domain = "https://nightly.wujciak.de/r" %}
+
+{{ macros.github_header() }}
+
+jobs:
+  source:
+    name: Source Package
+    runs-on: ubuntu-latest
+    outputs:
+      version: {{ '${{ steps.save-version.outputs.version }}' }}
+      date: {{ '${{ steps.save-version.outputs.date }}' }}
+    steps:
+      {{ macros.github_checkout_arrow()|indent }}
+      {{ macros.change_r_pkg_version()|indent }}
+      - name: Save Version
+        id: save-version
+        shell: bash
+        run: | 
+          echo "::set-output name=version::$(grep ^Version arrow/r/DESCRIPTION | sed s/Version:\ //)"
+          echo "::set-output name=date::$(date +%Y%m%d)"
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          install-r: false
+
+      - name: Build R source package
+        shell: bash
+        run: |
+          cd arrow/r
+          # Copy in the Arrow C++ source
+          make sync-cpp
+          R CMD build --no-build-vignettes .
+
+      - name: Install davfs2 & Mount Repo

```

Comment:
Instead of using webdav to test locally, how about a workflow that is like:

* jobs to build source and binary packages, store in github artifacts
* jobs to test source and binary installation, which start with steps (macro this out) that download the artifacts, put them in the right directories for an R repo, and use `python3 -m http.server $port` to serve the repo locally, and set `options(arrow.dev_repo = "http://localhost:$port")`
* if `nightly` (and those succeed), job to upload to webdav (and possibly test installing from there with `options(arrow.dev_repo = "https://nightlies.apache.org/arrow/r")` or whatever

We can't have PRs push to a remote nightly server to test.

Reply:
I never intended for this to be run on PR's, as you said using the remote server in this way would not do. I thought the intent of the way the arrow-r-nightly job was setup (with pushing to s3 in each step instead of using `actions/upload-artifact` ) was to test the proper creation of the artifacts *and* their upload to and the functionality of the nightly repo.

Sorry if I missed your intention to use this on PRs. I don't think that using this crossbow job on PRs would be necessary as we have `r.yml` which does a lot of similar things and has the benefits of being part of the check suit and running automatically instead of needing to trigger it manually. We could instead polish the `r.yml`-workflow e.g. by adding macos builds and other improvements (fix caching on the windows builds ...).

A small side note we wouldn't actually need to serve them via `python3`, you can use local dirs as repos ` install.packages("arrow", repos = "file:C:\\repo", type = "binary")`.

Reply:
p.s. I am using webdav for this because https://nightlies.apache.org uses it and compared to just pushing stuff over via curl it allows us to update the package index so more than just the latest package is available.

Reply:
It is a bit of scope extension to have this able to be requested + run on PRs, but I do think that is a feature that would be nice to have. It would allow us to test some of the setup scripts that are currently un(der)-tested.

And as for using a local filesystem versus a URL. Like ^^^ it would allow us to test source linux installs when people are getting libarrow as a binary, testing code like  https://github.com/apache/arrow/blob/442b24b0b9cf11d45245564fa7448f1ca4931ae7/r/tools/nixlibs.R#L63-L85 actually does what we want.

We can do all of this as a follow on if we prefer, of course.

Reply:
Done